tests/hikari/impl/test_event_manager_base.py

```diff
@@ -707,3 +729,18 @@ async def test(event: member_events.MemberCreateEvent):
                 ...
 
         subscribe.assert_called_once_with(member_events.MemberCreateEvent, test, _nested=1)
+
+    def test_listen_when_multiple_params_provided_as_union_in_typehint(self, event_manager):
```

Comment:
There should be a separate test case for 3.10+ style unions as these are a different class